src/collectors/collector_worker.go

```diff
@@ -93,13 +93,18 @@ func (cw *CollectorWorker) Run(device configuration.DeviceConfig) (string, error
 
 	// handle client timeouts
 	clientConnected := make(chan struct{})
-	go func() {
-		connection, err = utils.SSHClient(*s)
+
+	go func(err *error) {
+		connection, *err = utils.SSHClient(*s)
```

Comment:
This is a thing of beauty. I'm glad passing the err in worked.

Perhaps add a comment here that "close" closes the channel, returning to the select wait immediately. (To help our future selves from the confusion we had yesterday.)

Reply:
Added some comments